web/src/rest/publish.js

```diff
@@ -124,32 +134,19 @@ const publishRest = new Vue({
   },
 });
 
-// This has to be done with an interceptor because
-// the value of publishRest.token changes over time.
+// This is done with an interceptor because the value of
+// oauthClient.authHeaders is initialized asynchronously,
+// and doesn't exist at all if the user isn't logged in.
 // Using client.defaults.headers.common.Authorization = ...
-// would not update when the token does.
+// would not update when the headers do.
 client.interceptors.request.use((config) => {
-  if (!publishRest.token) {
-    return config;
-  }
   return {
     ...config,
     headers: {
-      Authorization: `Token ${publishRest.token}`,
+      ...oauthClient.authHeaders,
```

Comment:
If this runs only once, it should be fine. If this runs a second time (particularly on logout), then any existing auth token header will be carried forward and not actually removed. I think the library could start returning a null value in the header, but the order here of the spread values will have to switch.

Reply:
This is an [axios interceptor](https://github.com/axios/axios#interceptors) that injects the `oauthClient.authHeaders` individually into every request. I think this pattern is preferable to resetting `client.defaults.headers.common` every time `oauthClient.authHeaders` might have changed.